pydicom/valuerep.py

```diff
@@ -463,43 +455,49 @@ class IS(int):
     originally read or stored.
     """
 
-    def __new__(cls, val):
+    def __new__(
+        cls: Type[_IS], val: Union[None, str, int, float, Decimal]
+    ) -> Union[None, str, _IS]:
         """Create instance if new integer string"""
         if val is None:
             return val
+
         if isinstance(val, str) and val.strip() == '':
             return ''
 
-        newval = super(IS, cls).__new__(cls, val)
-
+        newval = super().__new__(cls, val)
         # check if a float or Decimal passed in, then could have lost info,
         # and will raise error. E.g. IS(Decimal('1')) is ok, but not IS(1.23)
         if isinstance(val, (float, Decimal)) and newval != val:
             raise TypeError("Could not convert value to integer without loss")
+
         # Checks in case underlying int is >32 bits, DICOM does not allow this
         check_newval = (newval < -2 ** 31 or newval >= 2 ** 31)
```

Comment:
Strange variable name - I would expect something like `is_out_of_range` (I know this is old code, I just find it difficult to read).